core/domain/exp_services.py

```diff
@@ -1293,17 +1298,17 @@ def update_version_history(
                 ),
                 committer_ids=updated_committer_ids
             ))
-        updated_version_history_model.update_timestamps()
-        updated_version_history_model.put()
+        return updated_version_history_model
 
 
-def _save_exploration(
+def _compute_models_to_put(
     committer_id: str,
     exploration: exp_domain.Exploration,
     commit_message: Optional[str],
     change_list: List[exp_domain.ExplorationChange]
-) -> None:
-    """Validates an exploration and commits it to persistent storage.
+) -> List[base_models.BaseModel]:
+    """Validates an exploration and returns a list of updated models related
```

Comment:
Can the validation happen in the caller? Seems out of place here given this function's name.

(If there are multiple callers, might need to modify the function's name.)